core-plugins/src/test/java/co/cask/hydrator/plugin/batch/action/HDFSActionTestRun.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.action;
+
+import co.cask.cdap.datapipeline.SmartWorkflow;
+import co.cask.cdap.etl.api.action.Action;
+import co.cask.cdap.etl.api.batch.BatchSink;
+import co.cask.cdap.etl.api.batch.BatchSource;
+import co.cask.cdap.etl.proto.v2.ETLBatchConfig;
+import co.cask.cdap.etl.proto.v2.ETLPlugin;
+import co.cask.cdap.etl.proto.v2.ETLStage;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.artifact.AppRequest;
+import co.cask.cdap.test.ApplicationManager;
+import co.cask.cdap.test.WorkflowManager;
+import co.cask.hydrator.plugin.batch.ETLBatchTestBase;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Test for {@link HDFSAction}
+ */
+public class HDFSActionTestRun extends ETLBatchTestBase {
+
+  @Test
+  public void testHDFSAction() throws Exception {
+    File file = new File("testFiles/source/test.txt");
```

Comment:
Should use `TemporaryFolder` `ClassRule` instead. It handles resource cleanup automatically, hence you don't need the `deleteDirectory` method. Also unit-test shouldn't create file in non-target directory.